e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,529 @@
+
+	// ACTION:- Configure an already configured Minter to another Minter Controller
+	// EXPECTED: Success; Minter Controller is configured with new Minter. Minter can have multiple Minter Controllers.
+	// Status:
+	// 	minterController1 -> minter3
+	// 	minterController2 -> minter2
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "minter-controller-3", math.OneInt(), noble)
+	minterController3 := w[0]
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "configure-minter-controller", minterController3.FormattedAddress(), minter3.FormattedAddress())
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMCRes, err = showMinterController(ctx, val, minterController3)
+	require.NoError(t, err, "failed to query show-minter-controller")
+	expectedShowMinterController = fiattokenfactorytypes.QueryGetMinterControllerResponse{
+		MinterController: fiattokenfactorytypes.MinterController{
+			Minter:     minter3.FormattedAddress(),
+			Controller: minterController3.FormattedAddress(),
+		},
+	}
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	showMCRes, err = showMinterController(ctx, val, minterController1)
+	require.NoError(t, err, "failed to query show-minter-controller")
+	expectedShowMinterController = fiattokenfactorytypes.QueryGetMinterControllerResponse{
+		MinterController: fiattokenfactorytypes.MinterController{
+			Minter:     minter3.FormattedAddress(),
+			Controller: minterController1.FormattedAddress(),
+		},
+	}
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	res, _, err := val.ExecQuery(ctx, "fiat-tokenfactory", "list-minter-controller")
+	require.NoError(t, err, "failed to query list-minter-controller")
+	var listMinterController fiattokenfactorytypes.QueryAllMinterControllerResponse
+	_ = json.Unmarshal(res, &listMinterController)
+	// ignore error because `pagination` does not unmarshal
+
+	expectedListMinterController := fiattokenfactorytypes.QueryAllMinterControllerResponse{
+		MinterController: []fiattokenfactorytypes.MinterController{
+			{ // this minter and controller were created/assigned at genesis
+				Minter:     nw.fiatTfRoles.Minter.FormattedAddress(),
+				Controller: nw.fiatTfRoles.MinterController.FormattedAddress(),
+			},
+			{
+				Minter:     minter3.FormattedAddress(),
+				Controller: minterController1.FormattedAddress(),
+			},
+			{
+				Minter:     minter2.FormattedAddress(),
+				Controller: minterController2.FormattedAddress(),
+			},
+			{
+				Minter:     minter3.FormattedAddress(),
+				Controller: minterController3.FormattedAddress(),
+			},
+		},
+	}
+
+	require.ElementsMatch(t, expectedListMinterController.MinterController, listMinterController.MinterController)
+}
+
+func TestFiatTFRemoveMinterController(t *testing.T) {
```

Comment:
missing happy path

Reply:
Added in: https://github.com/noble-assets/noble/pull/368/commits/a278e1e59be0319c50ef61a3a1ff79105c8fa702